### PR TITLE
[ENH] Fix inverted label mapping in OutlierDetectionAdapter & add adapter tests

### LINKED ISSUE #3184: [ENH] Add test coverage for anomaly_detection.collection estimators
### Describe the issue
While exploring the anomaly detection module (I'm interested in contributing to this area), I noticed that the collection submodule doesn't have any tests, while the series submodule has comprehensive test coverage.

Looking through the code, I found that _classification.py and _outlier_detection.py were added in PR #2326 (commit c8c63d10) and both appear to be fully testable, they even include _get_test_params() methods, which suggests tests were intended but never added.

**Current State
Missing:**

- aeon/anomaly_detection/collection/tests/ directory (doesn't exist)
- Test coverage for ClassificationAdapter
- Test coverage for OutlierDetectionAdapter

For comparison: aeon/anomaly_detection/series/tests/ has test_base.py and test_pyod_adapter.py with good coverage.

**Suggest a potential alternative/fix**
I'd like to add comprehensive tests for the collection module to bring it up to the same standard as the series module:

**Proposed Changes:**

Create test infrastructure:
- Add aeon/anomaly_detection/collection/tests/ directory
- Add __init__.py

Add test files:
- test_classification.py - Tests for ClassificationAdapter
- test_outlier_detection.py - Tests for OutlierDetectionAdapter

Test coverage will include:

- Initialization with valid and invalid parameters
- Fit and predict functionality
- Integration with sklearn/aeon estimators
- Error handling for edge cases
- Reproducibility checks

Additional context
No response

### PR DESCRIPTION
**Pull request by Adityakushwaha2006**
#### Reference Issues/PRs
Fixes #3184
Addresses the request for targeted test coverage for collection anomaly detection adapters.
Fixes a logic error discovered in `OutlierDetectionAdapter`.

#### What does this implement/fix? Explain your changes.

This PR adds unit tests for `ClassificationAdapter` and `OutlierDetectionAdapter` as requested and fixes a critical logic bug in the outlier adapter.

**1. Bug Fix in `OutlierDetectionAdapter`**
* **Issue:** The previous implementation incorrectly mapped outliers (`-1`) to `0` (Normal) and inliers (`1`) to `1` (Anomaly).
* **Fix:** Updated `_predict` in `aeon/anomaly_detection/collection/_outlier_detection.py` to correctly map `-1` to `1` (Anomaly).
    * *Old Logic:* `pred[pred == -1] = 0` (Incorrect)
    * *New Logic:* `return (pred == -1).astype(int)` (Correct: Outlier becomes Anomaly/True)

**2. Added Targeted Tests**
Added two new test files focusing exclusively on the adapter "translation" logic (avoiding generic shape/crash tests covered by `check_estimator`):
* `aeon/anomaly_detection/collection/tests/test_outlier_detection.py`: Verifies that Sklearn labels (`-1`, `1`) are correctly translated to Aeon standards (`1`, `0`). Includes a safety test to ensure no raw `-1` values leak into the output.
* `aeon/anomaly_detection/collection/tests/test_classification.py`: Verifies that the `ClassificationAdapter` acts as a transparent passthrough for binary labels.

#### Does your contribution introduce a new dependency? If yes, which one?

No.

#### Any other comments?

The tests use `DummyClassifier` and `DummyOutlierDetector` mocks to ensure they are fast, deterministic, and do not require training real models.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you **after** the PR has been merged.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

##### For new estimators and functions
- [ ] I've added the estimator/function to the online [API documentation](https://www.aeon-toolkit.org/en/latest/api_reference.html).
- [ ] (OPTIONAL) I've added myself as a `__maintainer__` at the top of relevant files and want to be contacted regarding its maintenance. Unmaintained files may be removed. This is for the full file, and you should not add yourself if you are just making minor changes or do not want to help maintain its contents.

##### For developers with write access
- [ ] (OPTIONAL) I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.

### CODE REVIEW

File: aeon/anomaly_detection/collection/tests/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+"""Tests for collection anomaly detection module."""
+
+__all__ = ["test_classification", "test_outlier_detection"]
```
Comment (SebastianSchmidl) [inline aeon/anomaly_detection/collection/tests/__init__.py]: We usually do not include an `__all__` for test directories because they do not represent an official API.

File: aeon/anomaly_detection/collection/tests/test_outlier_detection.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
+"""Adapter-specific tests for OutlierDetectionAdapter.
+
+Tests focus exclusively on adapter logic: label translation from
+sklearn/PyOD conventions (-1 for outlier, 1 for inlier) to aeon
+conventions (1 for anomaly, 0 for normal).
+"""
+
+__maintainer__ = ["SebastianSchmidl"]
```
Comment (SebastianSchmidl) [inline aeon/anomaly_detection/collection/tests/test_outlier_detection.py]: Please remove me as maintainer.

File: aeon/anomaly_detection/collection/tests/test_outlier_detection.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
+"""Adapter-specific tests for OutlierDetectionAdapter.
```
Comment (SebastianSchmidl) [inline aeon/anomaly_detection/collection/tests/test_outlier_detection.py:1]: Please also create
- a test with `IsolationForest` from sklearn to test interoperability and
- a test passing a wrong outlier detection estimator (not `OutlierMixin`) expecting a `ValueError` on fit and not on initialization.

File: aeon/anomaly_detection/collection/tests/test_outlier_detection.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        random_state=42,
+    )
+
+    adapter.fit(X)
+    predictions = adapter.predict(X)
+
+    np.testing.assert_array_equal(predictions, expected_output)
+
+
+def test_outlier_adapter_no_sklearn_labels_in_output():
```
Comment (SebastianSchmidl) [inline aeon/anomaly_detection/collection/tests/test_outlier_detection.py]: Not really necessary because all the inputs in this test are already covered by the other tests.

File: aeon/anomaly_detection/collection/tests/test_classification.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
+"""Adapter-specific tests for ClassificationAdapter.
+
+Tests focus exclusively on adapter logic: verifying classifier predictions
+pass through unchanged (no label translation needed).
+"""
+
+__maintainer__ = ["SebastianSchmidl"]
```
Comment (SebastianSchmidl) [inline aeon/anomaly_detection/collection/tests/test_classification.py]: Please remove my handle here.

File: aeon/anomaly_detection/collection/tests/test_classification.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
+"""Adapter-specific tests for ClassificationAdapter.
```
Comment (SebastianSchmidl) [inline aeon/anomaly_detection/collection/tests/test_classification.py]: Same here: Please also add a test with `RandomForestClassifier` and a test for a non-`ClassifierMixin`.

**Review by SebastianSchmidl — CHANGES_REQUESTED** [review]

File: aeon/anomaly_detection/collection/tests/test_classification.py [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    assert np.all(np.isin(predictions, [0, 1]))
+
+
+def test_classification_adapter_invalid_estimator_error_on_fit():
+    """Test that invalid estimator raises ValueError on fit, not init."""
+    # Invalid estimator (string)
+    invalid_classifier = "not_a_classifier"
+    adapter = ClassificationAdapter(classifier=invalid_classifier)
+
+    # Should NOT error during init - error comes on fit
```
Comment (SebastianSchmidl) [inline aeon/anomaly_detection/collection/tests/test_classification.py]:
```suggestion
    # Should NOT error during init - error comes on fit
    adapter = ClassificationAdapter(classifier=invalid_classifier)

```

File: aeon/anomaly_detection/collection/tests/test_outlier_detection.py [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+    assert np.all(np.isin(predictions, [0, 1]))
+
+
+def test_outlier_adapter_invalid_estimator_error_on_fit():
+    """Test that invalid estimator raises ValueError on fit, not init."""
+    # Invalid estimator (string)
+    invalid_detector = "not_an_estimator"
+    adapter = OutlierDetectionAdapter(detector=invalid_detector)
+
+    # Should NOT error during init - error comes on fit
```
Comment (SebastianSchmidl) [inline aeon/anomaly_detection/collection/tests/test_outlier_detection.py]:
```suggestion
    # Should NOT error during init - error comes on fit
    adapter = OutlierDetectionAdapter(detector=invalid_detector)

```

**Review by SebastianSchmidl — DISMISSED** [review]
Nice work 👍🏼 

Just some minor documentation improvements

**Review by SebastianSchmidl — DISMISSED** [review]
LGTM, but let's wait until @MatthewMiddlehurst had a chance to look at this as well. I'm not super sure that we do not already have some dummy fixtures for the tests.

File: aeon/anomaly_detection/collection/tests/test_classification.py [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+        self.classes_ = np.unique(y)
+        self.is_fitted_ = True
+        return self
+
+    def predict(self, X):
+        """Mock predict method returning binary class labels (0 or 1)."""
+        if self.forced_predictions is not None:
+            return np.array(self.forced_predictions)
+        n_samples = X.shape[0] if hasattr(X, "shape") else len(X)
+        return np.array([i % 2 for i in range(n_samples)])
```
Comment (MatthewMiddlehurst) [inline aeon/anomaly_detection/collection/tests/test_classification.py]: We have some mock/dummy classifiers already exist in `testing` and `classification` with deterministic output. I would rather use those if possible, or add the testing classifiers if they cannot.

File: aeon/anomaly_detection/collection/tests/test_classification.py [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+
+def test_classification_adapter_with_real_random_forest():
+    """Test ClassificationAdapter with real sklearn RandomForestClassifier.
+
+    This test verifies interoperability with actual sklearn estimators.
+    Uses aeon's SummaryClassifier wrapper to handle 3D collection data.
+    """
+    from sklearn.ensemble import RandomForestClassifier
+
+    from aeon.classification.feature_based import SummaryClassifier
```
Comment (MatthewMiddlehurst) [inline aeon/anomaly_detection/collection/tests/test_classification.py]: you can import these at the top, no need for them to be in the function

File: aeon/anomaly_detection/collection/tests/test_outlier_detection.py [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+        """Mock fit method."""
+        self.is_fitted_ = True
+        return self
+
+    def predict(self, X):
+        """Mock predict returning sklearn convention (-1 outlier, 1 inlier)."""
+        if self.forced_predictions is not None:
+            return np.array(self.forced_predictions)
+        n_samples = X.shape[0] if hasattr(X, "shape") else len(X)
+        return np.array([(-1 if i % 2 == 0 else 1) for i in range(n_samples)])
```
Comment (MatthewMiddlehurst) [inline aeon/anomaly_detection/collection/tests/test_outlier_detection.py]: This would be better as a private class as we dont want it used elsewhere.

**Review by MatthewMiddlehurst — CHANGES_REQUESTED** [review]
Good catch on the output, got the labels backwards!

If you go into the testing module we have some pre-made testing data which can be used here instead of creating it when we don't need to. This is pretty minor though, so can merge without it if no change is made.

File: aeon/anomaly_detection/collection/tests/test_outlier_detection.py [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+
+    np.testing.assert_array_equal(predictions, expected_output)
+
+
+def test_outlier_adapter_with_real_isolation_forest():
+    """Test OutlierDetectionAdapter with real sklearn IsolationForest.
+
+    This test verifies interoperability with actual sklearn estimators.
+    """
+    from sklearn.ensemble import IsolationForest
```
Comment (MatthewMiddlehurst) [inline aeon/anomaly_detection/collection/tests/test_outlier_detection.py]: same here with the import

File: aeon/anomaly_detection/collection/tests/test_classification.py (hunk 9 above)
Comment (Adityakushwaha2006) [reply aeon/anomaly_detection/collection/tests/test_classification.py]: The existing mocks inherit from base classes, but our adapters wrap sklearn estimators . These dummies are adapter specific , so imo we can keep these here

File: aeon/anomaly_detection/collection/tests/test_classification.py [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+import pytest
+from sklearn.base import BaseEstimator, ClassifierMixin
+from sklearn.ensemble import RandomForestClassifier
+
+from aeon.anomaly_detection.collection import ClassificationAdapter
+from aeon.classification.feature_based import SummaryClassifier
+from aeon.testing.data_generation import make_example_3d_numpy
+
+
+class DummyClassifier(BaseEstimator, ClassifierMixin):
```
Comment (SebastianSchmidl) [inline aeon/anomaly_detection/collection/tests/test_classification.py]:
```suggestion
class _DummyClassifier(BaseEstimator, ClassifierMixin):
```
should also be private, right?

**Review by SebastianSchmidl — DISMISSED** [review]

**Comment by MatthewMiddlehurst** [comment]
currently failing tests

**Comment by TonyBagnall** [comment]
hi, Im closing this as its not been touched for a while and is failing tests, feel free to reopen @Adityakushwaha2006
